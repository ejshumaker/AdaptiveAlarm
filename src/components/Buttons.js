import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { TouchableOpacity } from 'react-native';

const Buttons = (props) => {
  const {
    title, onPress, textColor, backgroundColor,
  } = props;
  return (
    <TouchableOpacity onPress={() => onPress()}>
      <Button style={{ backgroundColor }}>
        <Text style={{ fontFamily: 'RNSMiles-XBold', color: textColor }}>{title.toUpperCase()}</Text>
      </Button>
    </TouchableOpacity>
  );
};

Buttons.propTypes = {
  title: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
  textColor: PropTypes.string,
  backgroundColor: PropTypes.string,
};

Buttons.defaultProps = {
  textColor: 'white',
  backgroundColor: 'black',
};

const Button = styled.View`
  width: 186px;
  height: 40px;
  border-radius: 8px;
  justify-content: center;
  margin-bottom: 10px;
`;

const Text = styled.Text`
  font-size: 14px;
  text-align: center;
  text-transform: uppercase;
`;

export default Buttons;
