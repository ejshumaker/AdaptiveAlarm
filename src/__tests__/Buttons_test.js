/* eslint-disable no-undef */
/* eslint-disable import/first */
import 'react-native';
import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Buttons from '../components/Buttons';

// jest.mock('react-native-svg');

Enzyme.configure({ adapter: new Adapter() });

jest.setTimeout(10000);
let wrapper = null;
const onPressMock = jest.fn();


describe('Buttons tests', () => {
  beforeEach(() => {
    wrapper = mount(<Buttons
      onPress={onPressMock}
      title="test"
      textColor="b"
      backgroundColor="w"
    />);
    jest.clearAllMocks();
  });

  it('test Buttons matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
