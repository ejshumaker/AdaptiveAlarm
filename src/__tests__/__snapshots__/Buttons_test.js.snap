// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons tests test Buttons matches snapshot 1`] = `
<Buttons
  backgroundColor="w"
  onPress={[MockFunction]}
  textColor="b"
  title="test"
>
  <TouchableOpacity
    onPress={[Function]}
  >
    <react-native-mock>
      <Styled(View)
        style={
          Object {
            "backgroundColor": "w",
          }
        }
      >
        <StyledNativeComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "AccessibilityComponentType": Array [
                "none",
                "button",
                "radiobutton_checked",
                "radiobutton_unchecked",
              ],
              "AccessibilityTraits": Array [
                "none",
                "button",
                "link",
                "header",
                "search",
                "image",
                "selected",
                "plays",
                "key",
                "text",
                "summary",
                "disabled",
                "frequentUpdates",
                "startsMedia",
                "adjustable",
                "allowsDirectInteraction",
                "pageTurn",
              ],
              "attrs": Array [],
              "displayName": "Styled(View)",
              "forceTouchAvailable": false,
              "inlineStyle": InlineStyle {
                "rules": Array [
                  "
  width: 186px;
  height: 40px;
  border-radius: 8px;
  justify-content: center;
  margin-bottom: 21px;
",
                ],
              },
              "render": [Function],
              "styledComponentId": "StyledNativeComponent",
              "target": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
          style={
            Object {
              "backgroundColor": "w",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderRadius": 8,
                  "height": 40,
                  "justifyContent": "center",
                  "marginBottom": 21,
                  "width": 186,
                },
                Object {
                  "backgroundColor": "w",
                },
              ]
            }
          >
            <react-native-mock>
              <Styled(Text)
                style={
                  Object {
                    "color": "b",
                    "fontFamily": "RNSMiles-XBold",
                  }
                }
              >
                <StyledNativeComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "displayName": "Styled(Text)",
                      "inlineStyle": InlineStyle {
                        "rules": Array [
                          "
  font-size: 14px;
  text-align: center;
  text-transform: uppercase;
",
                        ],
                      },
                      "render": [Function],
                      "styledComponentId": "StyledNativeComponent",
                      "target": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  style={
                    Object {
                      "color": "b",
                      "fontFamily": "RNSMiles-XBold",
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "fontSize": 14,
                          "textAlign": "center",
                          "textTransform": "uppercase",
                        },
                        Object {
                          "color": "b",
                          "fontFamily": "RNSMiles-XBold",
                        },
                      ]
                    }
                  >
                    <react-native-mock>
                      TEST
                    </react-native-mock>
                  </Text>
                </StyledNativeComponent>
              </Styled(Text)>
            </react-native-mock>
          </View>
        </StyledNativeComponent>
      </Styled(View)>
    </react-native-mock>
  </TouchableOpacity>
</Buttons>
`;
